In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("bridges.csv")

In [3]:
df.head()

,Id,river,location,erected,purpose,length,lanes,clear-g,t-or-d,material,span,rel-l,type
0,E1,M,3.0,1818,HIGHWAY,NaN,2.0,N,THROUGH,WOOD,SHORT,S,WOOD
1,E2,A,25.0,1819,HIGHWAY,1037.0,2.0,N,THROUGH,WOOD,SHORT,S,WOOD
2,E3,A,39.0,1829,AQUEDUCT,NaN,1.0,N,THROUGH,WOOD,NaN,S,WOOD
3,E5,A,29.0,1837,HIGHWAY,1000.0,2.0,N,THROUGH,WOOD,SHORT,S,WOOD
4,E6,M,23.0,1838,HIGHWAY,NaN,2.0,N,THROUGH,WOOD,NaN,S,WOOD


In [4]:
df.columns

Index(['Id', 'river', 'location', 'erected', 'purpose', 'length', 'lanes',
       'clear-g', 't-or-d', 'material', 'span', 'rel-l', 'type'],
      dtype='str')

In [5]:
df["length"].describe()

count      81.000000
mean     1567.469136
std       747.491523
min       804.000000
25%      1000.000000
50%      1300.000000
75%      2000.000000
max      4558.000000
Name: length, dtype: float64

I assume that the length shouldn't be extermly short or long ,which mean I consider saftey and convience ,So i will flage the exterm long and short.which is (804.4558).I will screen the length column for values that appearextermly short or extermly long based on my chosen range.

In [6]:
flagged = df[(df["length"] < 870) | (df["length"] > 2822)]
flagged

,Id,river,location,erected,purpose,length,lanes,clear-g,t-or-d,material,span,rel-l,type
31,E34,O,41.0,1888,RR,4558.0,2.0,G,THROUGH,STEEL,LONG,F,SIMPLE-T
44,E46,A,37.0,1897,RR,4000.0,2.0,G,DECK,STEEL,LONG,F,SIMPLE-T
52,E87,A,35.0,1903,RR,3000.0,2.0,G,THROUGH,STEEL,MEDIUM,S,SIMPLE-T
72,E70,A,27.0,1926,HIGHWAY,860.0,4.0,G,THROUGH,STEEL,MEDIUM,S-F,SUSPEN
80,E71,A,25.0,1928,HIGHWAY,860.0,4.0,G,THROUGH,STEEL,MEDIUM,S-F,SUSPEN
88,E107N,A,39.7,1945,RR,840.0,2.0,G,THROUGH,STEEL,MEDIUM,S-F,SIMPLE-T
98,E82,O,42.0,1955,HIGHWAY,804.0,NaN,G,THROUGH,STEEL,NaN,F,SIMPLE-T
104,E91,O,44.0,1975,HIGHWAY,3756.0,6.0,G,THROUGH,STEEL,LONG,F,ARCH


In [7]:
flagged[["length"]].sort_values("length")

,length
98,804.0
88,840.0
72,860.0
80,860.0
52,3000.0
104,3756.0
44,4000.0
31,4558.0


Handling the Flagged Values

I found 8 bridge lengths that were flagged which is extreme values. the range between( 804 to 4558). These values are exctreme compared with the rest of the data,whoever, it is not impossible value . I will keep these rows in the dataset because I do not have enough evidence to say that the values are true .it might shown for anyother reason or constraction issue which I don't have details about it. I will document them as  extreme values for further review.
